lib/db/leveldb_dbinstance.go

```diff
@@ -606,6 +659,23 @@ func (db *Instance) updateSchema0to1() {
 	l.Infof("Updated symlink type for %d index entries and added %d invalid files to global list", symlinkConv, ignAdded)
 }
 
+func (db *Instance) updateSchema1to2() {
+	t := db.newReadWriteTransaction()
+	defer t.close()
+
+	var sk []byte
+	var dk []byte
+
+	for _, folderStr := range db.ListFolders() {
+		folder := []byte(folderStr)
+		db.withHave(folder, protocol.LocalDeviceID[:], nil, true, func(f FileIntf) bool {
+			t.Put(db.sequenceKeyInto(sk, folder, f.SequenceNo()), db.deviceKeyInto(dk[:cap(dk)], folder, protocol.LocalDeviceID[:], []byte(f.FileName())))
```

Comment:
This is quite compact. It wouldn't kill you to make the key creation business a couple of lines of their own.